lib/board.rb

```diff
@@ -0,0 +1,45 @@
+class Board
+  attr_accessor :cells
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def initialize
+    reset!
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(pos)
+    @cells[pos.to_i-1]
+  end
+
+  def full?
+    @cells.all?{|full| full == "X" || full == "O"}
+  end
+
+  def turn_count
+    @cells.count {|full| full == "X" || full == "O"}
+  end
+
+  def taken?(pos)
+    position(pos) == "X" || position(pos) == "O" ? true:false
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 
```
def taken?(pos)
  position(pos) == "X" || position(pos) == "O"
end
```